osu.Game/Skinning/SkinProvidingContainer.cs

```diff
@@ -170,10 +208,10 @@ protected override void Dispose(bool isDisposing)
             base.Dispose(isDisposing);
 
             if (fallbackSource != null)
-                fallbackSource.SourceChanged -= TriggerSourceChanged;
+                fallbackSource.SourceChanged -= OnSourceChanged;
 
-            if (skin is ISkinSource source)
-                source.SourceChanged -= TriggerSourceChanged;
+            foreach (var source in SkinSources.OfType<ISkinSource>())
+                source.SourceChanged -= OnSourceChanged;
```

Comment:
Same here, I would actually like to see the event unbound from the list, just so that shenanigans don't occur if the list is changed after disposal.

Note also that this is effectively the same as just clearing the list.

Reply:
Probably needed to write a comment, but doing this may actually a "collection was modified; enumeration operation may not execute" exception due to the disposal being on a separate thread (at least that's what I had speculated from the test failure I reported above this review)

Reply:
Link here https://github.com/ppy/osu/pull/13423#issuecomment-858675220